megamolbart/launch.py

```diff
@@ -45,64 +45,54 @@ def __init__(self):
-        parser.add_argument('-c', '--checkpoints_dir',
-                            dest='checkpoints_dir',
-                            default=CHECKPOINTS_DIR,
-                            help='Path to Checkpoints dir.')
-
-        parser.add_argument('--num_layers',
-                            dest='num_layers',
-                            type=int,
-                            default=DEFAULT_NUM_LAYERS,
-                            help='Number hidden layers in the model')
-
-        parser.add_argument('--hidden_size',
-                            dest='hidden_size',
-                            type=int,
-                            default=DEFAULT_D_MODEL,
-                            help='Hidden size of the model')
-
-        parser.add_argument('--num_attention_heads',
-                            dest='num_attention_heads',
-                            type=int,
-                            default=DEFAULT_NUM_HEADS,
-                            help='Maximum length of decoded sequence')
+        parser.add_argument('-m', '--model_path',
+                            dest='model_path',
+                            default=None,
+                            help='Path to model content.')
 
         args = parser.parse_args(sys.argv[1:])
 
         if args.debug:
             logger.setLevel(logging.DEBUG)
 
-        logger.info(f'Maximum decoded sequence length is set to {args.max_decode_length}')
+        if args.model_path is None:
+            model_dir = self._fetch_model_path()
+        else:
+            model_dir = args.model_path
+        logger.info(f'Using checkpoint: {model_dir}')
 
         server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
         generativesampler_pb2_grpc.add_GenerativeSamplerServicer_to_server(
-            GenerativeSampler(decoder_max_seq_len=args.max_decode_length,
-                              vocab_path=args.vocab,
-                              checkpoints_dir=args.checkpoints_dir,
-                              num_layers=args.num_layers,
-                              hidden_size=args.hidden_size,
-                              num_attention_heads=args.num_attention_heads),
+            GenerativeSampler(model_dir=model_dir),
             server)
         server.add_insecure_port(f'[::]:{args.port}')
         server.start()
         server.wait_for_termination()
 
+    def _fetch_model_path(self, search_loc='/models'):
+        """
+        Fetch the model path from the model server.
+        """
+        checkpoints = sorted(pathlib.Path(search_loc).glob('**/megamolbart_checkpoint.nemo'))
+        logger.info(f'Found {len(checkpoints)} checkpoints in {search_loc}')
+        
+        if not checkpoints or len(checkpoints) == 0:
+            raise Exception('Model not found')
+        else:
+            checkpoint_dir = checkpoints[-1].absolute().parent.as_posix()
+
+            # TODO: This is a hack to place the vocab file where the model is expecting it.
```

Comment:
@rilango I think this is already where it's installed in the Docker container. Did you verify that it's needed? 

The code is OK either way, just wanted to note it.

Reply:
In the current way we download model while setup. We can do that while building container, but I stuck to our current workflow.